vyper/semantics/analysis/module.py

```diff
@@ -414,7 +414,7 @@ def visit_ImplementsDecl(self, node):
             raise StructureException(msg, node.annotation, hint=hint)
 
         # grab exposed functions
-        funcs = self._exposed_functions
+        funcs = self._all_functions
```

Comment:
i think `funcs = [fn_t for fn_t in self._all_functions if fn_t.is_external]`

Reply:
addressed in: https://github.com/vyperlang/vyper/pull/4178/commits/e4f6622cd729eb7cdd26e6c8e8c455082f4500ad